### ISSUE TITLE
Need to tab twice when working on an iPad

### ISSUE THREAD

**Issue by Reneeeee** [wekan/wekan#1123]
Hi, I'm new here and love WeKan (really meets Trello quite well).

I did experience some inconveniences though:

1) When adding/editing a new/existing project, task or subtask or when trying to select a blanc textfield on an iPad, one need to tab twice instead of once.

2) The checklist and date options are only available in existing tasks/subtasks. Not when creating/adding.

Both issues make it a bit irritating to work with WeKan on an iPad. Could those be corrected please?

I will try to work on the Dutch translation in the nest future, but have to dig into the manual first (am a designer with limited technical skills :)

Greetz, Rene

**Comment by Reneeeee** [comment]
Thanks xet7!

Maybe topic should be moved to wefork/wekan since I run WeKan on Sandstorm (I'm sorry for not posting it their in the first place...)?

**Comment by xet7** [comment]
@Reneeeee 

I have merged Wefork back to Wekan previously:
https://github.com/wekan/wekan/wiki/FAQ#what-was-wekan-fork--wefork

Currently Sandstorm has old version of Wekan, after that there has been lots of changes. Anyway at some point of time Sandstorm will get latest version of Wekan.

I will test does this bug happen on standalone source/Docker version of Wekan.

**Comment by Reneeeee** [comment]
Thanks for the effort and information xet7! 

I tested it in the original [Sandstorm demo](https://oasis.sandstorm.io/demo). I'm not sure if I had installed other app's (grains) during that test. If you want me to do a new test only running WeKan in Sandstorm, let me know and I will fire it on my iPad (lies two floors lower, am now working on my iMac).

Really hope the latest WeKan will be integrated into the next release of Sandstorm...

**Comment by ocdtrekkie** [comment]
@Reneeeee I can tag @dwrensha here about that, but the main reason the Sandstorm package lags behind is there's extra work in packaging it, and everything needs to be tested that it works correctly inside the Sandstorm sandbox.

Sandstorm's packaged version is arguably the most secure way to use Wekan between the general sandboxing behavior, and that there are security bugs that have been introduced to Wekan and fixed since the last Sandstorm package.

**Comment by xet7** [comment]
@ocdtrekkie

What do you think, is it ok to release new version of Wekan although #1105 is not fixed yet? Fixing it would require import of Wekan board to handle attachments as base64 encoded into JSON, and then preventing access to files without authorization.

I don't know does current version of capnp work with current version of Wekan.

**Comment by ocdtrekkie** [comment]
@xet7 I doubt #1105 would affect Sandstorm. Only users who have been shared access to the Wekan grain can access files within it due to the Sandstorm sandbox. And since each grain contains only a single board (of which everyone it's shared with can at least view it), I do not think it would be possible for unauthorized users to access that data within Sandstorm.

**Comment by xet7** [comment]
@ocdtrekkie

Ok thanks, I'll release new version of Wekan. I also would like to get import/export of of Wekan board with attachments working in Sandstorm as it's blocking some deployments of Sandstorm. I'll look at it at some point of time with @GhassenRjab .

**Comment by Reneeeee** [comment]
Thanks for clarification @ocdtrekkie and thanks for following up again @xet7. Will continue following this thread.

**Comment by ocdtrekkie** [comment]
@xet7 I wouldn't release a new version that non-Sandstorm users update to if that bug is introduced by the current version. I assume that we still need David to publish new Sandstorm releases at this time?

**Comment by xet7** [comment]
@ocdtrekkie

I presume that bug has been on Wekan always, I don't know what version has introduced it. It's just that someone found it recently. The same bug is still at Trello, that's how importing attachments to Wekan work, although I have not checked the code yet.

**Comment by ocdtrekkie** [comment]
Got it. For some reason I thought it was introduced with the API or something like that.

Some would argue the file URL's being unguessable is good enough security to begin with for most users. If they are, in fact, unguessable, the only problem is that one a user who has access has the URL, they can share it with others, and you can't revoke that user's access to that URL even if you remove them from the Wekan board. That's a very specific risk.

**Comment by xet7** [comment]
@ocdtrekkie

Another alternative is to wait that:
- base64 attachments to JSON are implemented
- authentication is added to standalone Wekan's files
- Wekan import and export can be made visible at Sandstorm

And then get it working on Sandstorm.

**Comment by xet7** [comment]
Moved to #953
